### PR TITLE
Add option to disable replacing 0 with %s when creating arrays

### PR DESCRIPTION
**Pull request by 9names**
When creating an array out of registers we take the first register's description and replace any 0's with "%s".
This might be the best choice for a lot of registers, but there's also cases where this new description makes no sense.
This PR preserves the previous behavior as the default, but also provides an option for users to preserve the original description.

See https://github.com/rp-rs/rp2040-pac/pull/39/commits/bcf98e8cc6c745cf7cc50f9f4d8d13c14a7e9a81 for an example of where the current implementation results in nonsense output.

### CODE REVIEW

File: svdtools/patch.py [hunk 1]
```diff
@@ -912,7 +912,7 @@ def collect_in_array(self, rspec, rmod):
             name = rmod["name"]
         else:
             name = rspec[:li] + "%s" + rspec[len(rspec) - ri :]
-        if dimIndex[0] == "0":
+        if not rmod.get("_keep_zero_character", "") and dimIndex[0] == "0":
```
Comment (adamgreig) [inline svdtools/patch.py:915]:
```suggestion
        if not rmod.get("_keep_zero_character") and dimIndex[0] == "0":
```

The default is `None` which should evaluate false-y here like `""`, right?

Comment (9names) [reply svdtools/patch.py:915]: Right you are, it is indeed None which is truthy!
Want me to make the same change to the code I used as an example?
https://github.com/stm32-rs/svdtools/blob/7c4bf815b8ba185cb1e92e8e1a4a0c3e1ee46192/svdtools/patch.py#L886

Comment (9names) [reply svdtools/patch.py]: I've assumed that you'd want the line 886/1236 changes as well in commit `0dda853 `, let me know if you want me to amend to just fix the new code.

**Review by adamgreig — APPROVED** [review]
Thanks!

bors merge

**Comment by adamgreig** [comment]
bors retry

**Pull request merged**